test/setVersionByAcceptHeader.test.js

```diff
@@ -0,0 +1,162 @@
+
+  t.context.req.headers['accept'] = 'application/vnd.company+json; param1=1,      Version =' + versionNumber + '  , param3=3'
+  const middleware = versionRequest.setVersionByAcceptHeader()
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, versionNumber)
+  })
+})
+
+test('dont set the version if the Accept header has no "version" parameter', t => {
+  t.context.req.headers['accept'] = 'application/vnd.company+json;param1=1, param2=2'
+  const middleware = versionRequest.setVersionByAcceptHeader()
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, undefined)
+  })
+})
+
+test('dont set the version if the Accept header has no parameters at all', t => {
+  t.context.req.headers['accept'] = 'application/vnd.company+json;'
+  const middleware = versionRequest.setVersionByAcceptHeader()
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, undefined)
+  })
+})
+
+test('dont set the version if the Accept header has no parameters at all (without ending ;)', t => {
+  t.context.req.headers['accept'] = 'application/vnd.company+json'
+  const middleware = versionRequest.setVersionByAcceptHeader()
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, undefined)
+  })
+})
+
+//  Alternative format
+
+test('we can set the version using the Accept header alternative format', t => {
+  const versionNumber = '1.0.0'
+
+  t.context.req.headers['accept'] = 'application/vnd.company-v' + versionNumber + '+json'
+  const middleware = versionRequest.setVersionByAcceptHeader()
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, versionNumber)
+  })
+})
+
+test('we can set the version using the Accept header alternative format, even if it has whitespaces', t => {
+  const versionNumber = '1.0.0'
+
+  t.context.req.headers['accept'] = 'application/ vnd.company -v' + versionNumber + ' + json'
+  const middleware = versionRequest.setVersionByAcceptHeader()
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, versionNumber)
+  })
+})
+
+test('dont set the version, if the alternative format is incorrect', t => {
+  const versionNumber = '1.0.0'
+
+  t.context.req.headers['accept'] = 'application/vnd.company+v' + versionNumber + '/json'
+  const middleware = versionRequest.setVersionByAcceptHeader()
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, undefined)
+  })
+})
+
+//  Custom function
+
+test('we can set the version using a custom function to parse the Accept header', t => {
+  const versionNumber = '1.0.0'
+
+  t.context.req.headers['accept'] = versionNumber
+  const middleware = versionRequest.setVersionByAcceptHeader(v => v)
+
+  middleware(t.context.req, {}, () => {
+    t.deepEqual(t.context.req.version, versionNumber)
+  })
+})
+
+test('we can handle, if the custom function returns a number', t => {
+  const versionNumber = '1.1'
```

Comment:
this case will work but probably an accept header with just a value of '1.1' is not RFC compliant. we can leave the test as is but I wanted to point it out.